### ISSUE TITLE
[deploy] `No deployers were triggered` when running :jreleaserDeploy

### ISSUE THREAD

**Issue by pacu** [jreleaser/jreleaser#1736]
I'm trying to publish to maven central new portal. I have completed all the steps (portal account, gpg keys, environment variables, etc) yet I'm still not achieving a successful deploy.

### Task List

- [x] Steps to reproduce provided
- [ ] Stacktrace (if present) provided
- [x] Example that reproduces the problem (link to git repository is ideal)
- [x] Full description of the issue provided (see below)

### Steps to Reproduce
Checkout https://github.com/zecdev/zcash-kotlin-payment-uri/pull/31
1. ./gradlew :lib:publish
2. ./gradlew :lib:jreleaserDeploy

build.kts

https://github.com/zecdev/zcash-kotlin-payment-uri/pull/31/files#diff-70c8e4a4271e469be33be7a8b27a91caf0161786d33398b57af09813697b2c56

### Expected Behaviour

It should deploy to maven central 

### Actual Behaviour

````
 Task :zip321:jreleaserPublish
[INFO]  JReleaser 1.14.0
[INFO]    - basedir set to [path]/Zcash/zcash-kotlin-payment-uri/zip321
[INFO]    - outputdir set to[path]/Zcash/zcash-kotlin-payment-uri/zip321/build/jreleaser
[INFO]  git-root-search set to true
[INFO]  Loading variables from [USER]/.jreleaser/config.toml
[INFO]  Validating configuration
[INFO]  Strict mode set to false
[INFO]  Project version set to 0.0.3-SNAPSHOT
[INFO]  Release is snapshot
[INFO]  Timestamp is 2024-10-03T21:21:20.92574-03:00
[INFO]  HEAD is at cdce4d6
[INFO]  Platform is osx-x86_64
[INFO]  dry-run set to false
[INFO]  Calculating checksums for distributions and files
[INFO]    [checksum] No files configured for checksum. Skipping
[INFO]  Writing output properties to build/jreleaser/output.properties
[INFO]  JReleaser succeeded after 0.003 s
````

### Environment Information

- **Operating System**: MacOS 17.4 (Sonoma)

- **JReleaser Version:** 1.14
- **JDK Version:** 21

[trace.log](https://github.com/user-attachments/files/17251533/trace.log)

**Comment by aalmiray** [comment]
Not a bug per-se but the output could be more precise. No deployers triggered because:

- the project version was resolved as snapshot (`0.0.3-SNAPSHOT`)
```
project:
    name: zip321
    version: 0.0.3-SNAPSHOT
```
- the new maven central publishing APIs do not allow snapshot publications. However, that piece of information is missing in the docs. Compare https://jreleaser.org/guide/latest/reference/deploy/maven/maven-central.html vs. https://jreleaser.org/guide/latest/reference/deploy/maven/azure.html
- the mavenCentral deployer was disabled as a result
```
[DEBUG]   [validation] deploy
[DEBUG]   [validation] deploy.maven
[DEBUG]   [validation] deploy.maven.mavenCentral
[DEBUG]   [validation] deploy.maven.mavenCentral.maven-central
[DEBUG]   [validation] - disabled
```

Action items:
- add a warning to the output indicating why the deployer was disabled
- document that mavenCentral deployer does not support snapshot publication

**Comment by aalmiray** [comment]
🎉 This issue has been resolved in `v1.15.0` ([Release Notes](https://github.com/jreleaser/jreleaser/releases/tag/v1.15.0))
